### DISCUSSION TITLE
WindowsAppRuntime with lower version will not install if higher version already installed

### DISCUSSION THREAD

**Discussion by rcohn** [microsoft/WindowsAppSDK#2714] [General]
Hi -

We are creating a C# WinUI3 desktop unpackaged app built with WindowsAppSDK 1.0.0.  

The installer for this app copies the app artifacts (from the set built in proj\bin\x64\release\net5.0-windows.10.0.19041.0\win10-x64) to a target location.  Part of the installation involves installing the version of WindowsAppRuntime that corresponds to the version of WindowsAppSDK used to build the app (in this case 1.0.0).  A Start menu link is created to point to the app exe file.

_**This works OK, except when a higher version of WindowsAppRuntime is already installed****, in which case the WindowsAppRuntime will not be installed, and it complains about the higher version.**_

In this case, seemingly the only recourse is to uninstall the existing WindowsAppRuntime packages, which is a tedious task and something that we can't expect the final user to perform.

Is there a way to install all of the necessary runtime components needed that isolates the app from any existing WindowsAppRuntime installed components?  I did some quick testing with `<WindowsAppSDKSelfContained>` and, while it worked for a simple app built in the same way, it did not work for the product app (app crashed on startup).

I hope that you can give me some advice on the best way to install such an unpackaged WinUI3 app.

Thanks very much.

**Comment by DrusTheAxe** [comment]
This is a dupe of [WindowsAppRuntime installation error: The package could not be installed because a higher version of this package is already installed. #2708](https://github.com/microsoft/WindowsAppSDK/discussions/2708) ?
